### DISCUSSION TITLE
RTMP HTML5 Video player using HLS with ngrok URL

### DISCUSSION THREAD

**Discussion by PatchaSravani2** [bluenviron/mediamtx#2707] [Questions] [answered]
### Question

How to view RTMP live stream in web browser using HLS using ngrok?
I'm using the following ffmpeg command to start the stream
ffmpeg -i /dev/video0 -c:v libx264 -tune zerolatency -g 30 -b:v 100k -maxrate 100k -bufsize 200k -c:a aac -b:a 96k -ar 22050 -s 426x240 -r 24 -f flv "rtmp://localhost:1935/mystream/{Drone_code}_{mission_uuid}?user=admin&pass=admin"

If I use localhost it is working fine, but the problem comes if I use ngrok URL. index.m3u8 is loading but the live stream video is not playing in browser.

**Comment by aler9** [comment, marked as answer]
Hello, both RTMP and HLS work by using a single TCP port each, that are respectively 1935 and 8888.

If you want to send a stream with RTMP through ngrok or any tunnel, make sure that requests to port 1935 are forwarded to your server.
If you want to read a stream with HLS through ngrok or any other tunnel, make sure that requests to port 8888 are forwarded to your servers.

No additional steps are required.

**Reply by PatchaSravani2** [reply]
Thank you.

[a comment by an automated account is not shown]
